pandas/core/arrays/datetimelike.py

```diff
@@ -62,7 +61,7 @@ def _simple_new(cls, values, **kwargs):
         raise AbstractMethodError(cls)
 
     @property
-    def _scalar_type(self) -> Type[DatetimeLikeScalar]:
+    def _scalar_type(self) -> Type[Union[Period, Timestamp, Timedelta]]:
```

Comment:
why is this changing?

Reply:
fix for pandas\core\arrays\datetimes.py:275: error: Incompatible types in assignment (expression has type "Type[Timestamp]", base class "AttributesMixin" defined the type as "Type[DatetimeLikeScalar]") arising from adding type information for Timestamp.

Reply:
DatetimeLikeScalar is a typevar.

Reply:
In _typing.py
```
AnyArrayLike = TypeVar(
    "AnyArrayLike", "ExtensionArray", "Index", "Series", "SparseSeries", np.ndarray
)
ArrayLike = TypeVar("ArrayLike", "ExtensionArray", np.ndarray)
DatetimeLikeScalar = TypeVar("DatetimeLikeScalar", "Period", "Timestamp", "Timedelta")
Dtype = Union[str, np.dtype, "ExtensionDtype"]
```

these all resolve to Any due to missing imports, so probably hiding many errors.

Reply:
yes. that's the idea, then after the skeleton stubs are added and any issues arising are sorted, we can start adding type information to the stubs.

using --disallow-any-expr on module identifies where type information is needed otherwise gradually type like the rest of the codebase.

Reply:
yes. good catch. have updated period.py and timedeltas.py to match.